apps/AEPSampleApp/extensions/EdgeView.tsx

```diff
@@ -43,29 +102,45 @@ const EdgeView = ({navigation}: NavigationProps) => {
        locationStr = "null";
       }
       getlocationHintText(locationStr);
-  })
+    });
   }
 
   return (
     <View style={styles.container}>
       <ScrollView contentContainerStyle={{marginTop: 75}}>
         <Button onPress={() => navigation.goBack()} title="Go to main page" />
         <Text style={styles.welcome}>Edge v{version}</Text>
-        <Button title="sendEvent()" onPress={() => sendEvent()} />
         <Button
-          title="sendEvent() to Dataset"
-          onPress={() => sendEvent('datasetIdExample')}
+          title="sendEventWithParams"
+          onPress={() => sendEventWithParams()}
```

Comment:
I think we do arrow functions all over, but this can just be `onPress={sendEventWithParams}`

Reply:
I am not able to remove the arrow without getting errors.
Possible Unhandled Promise Rejection (id: 0):
Error: Exception in HostFunction: Malformed calls from JS: field sizes are different.

Could the way how the function was written
```
 function sendEventWithParams(datasetId?: string) {
    const sampleXdmData = {eventType: 'SampleXDMEvent'};
    const freeFormData = {free: 'form', data: 'example'};

    // Method using params
    let experienceEvent = new ExperienceEvent(sampleXdmData, freeFormData, datasetId);

    sendEvent(experienceEvent);
  }
 ```
 
 This is in the sample app, I am thinking to leave it as is.